game/ui/general/transactioncontrol.h

```diff
@@ -94,6 +94,18 @@ class TransactionControl : public Control
 		int getBalance() const { return getRightStock(true); }
 		// ScrollBar support. Set current value.
 		int setBalance(const int balance);
+
+		/// <summary>
+		/// Checks if trade is active or not. Can be used to determine to draw shades, remove from
+		/// item list, etc.
+		/// </summary>
+		/// <returns>If trade is active or not.</returns>
+		bool isActive() const
+		{
+			auto isActive = (getLeftIndex() == getRightIndex() ||
```

Comment:
Maybe this logic can be split out?

if (getLeftIndex() == getRightIndex())
  return false;
if (getLeftStock == 0 && getRightStock() == 0)
  return false;
return true;

Reply:
Logic improved

https://github.com/OpenApoc/OpenApoc/blob/f8c0609d18e468c19a91894ab58f33132f74d01c/game/ui/general/transactioncontrol.h#L103-L112